src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,423 @@
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.ServerSocket;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+import net.snowflake.client.RunningNotOnGithubActionsMac;
+import net.snowflake.client.RunningNotOnJava21;
+import net.snowflake.client.RunningNotOnJava8;
+import net.snowflake.client.category.TestCategoryOthers;
+import net.snowflake.client.core.HttpUtil;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(TestCategoryOthers.class)
+public class ProxyLatestIT {
+
+  private static final Logger logger = Logger.getLogger(ProxyLatestIT.class.getName());
+
+  private static final String WIREMOCK_HOME_DIR = ".wiremock";
+  private static final String WIREMOCK_FILE_NAME = "wiremock-standalone-3.8.0.jar";
+  public static final String WIREMOCK_STANDALONE_URL =
+      "https://repo1.maven.org/maven2/org/wiremock/wiremock-standalone/3.8.0/wiremock-standalone-3.8.0.jar";
+  private static final String WIREMOCK_HOST = "localhost";
+  private static final String TRUST_STORE_PROPERTY = "javax.net.ssl.trustStore";
+  private static int httpProxyPort;
+  private static int httpsProxyPort;
+  private static String originalTrustStorePath;
+  private static Process wiremockStandalone;
+
+  @BeforeClass
+  public static void setUpClass() {
+    assumeFalse(RunningNotOnJava8.isRunningOnJava8());
+    assumeFalse(RunningNotOnJava21.isRunningOnJava21());
+    assumeFalse(
+        RunningNotOnGithubActionsMac
+            .isRunningOnGithubActionsMac()); // disabled until issue with access to localhost is
+    // fixed on github actions mac image
+    originalTrustStorePath = systemGetProperty(TRUST_STORE_PROPERTY);
+    downloadWiremock();
+    startWiremockStandAlone();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    System.setProperty(
+        TRUST_STORE_PROPERTY, getResourceURL("wiremock" + File.separator + "ca-cert.jks"));
```

Comment:
do we need to do such setup before each test or it can be done at the beginning of the whole class?

Reply:
I changed to execute it inside tests that needs that